krynn/krynn.php

```diff
@@ -0,0 +1,174 @@
+
+	$active = get_pconfig(local_user(), 'krynn', 'enable');
+
+	if(! $active)
+		return;
+
+	/**
+	 *
+	 * OK, we're allowed to do our stuff.
+	 * Here's what we are going to do:
+	 * load the list of timezone names, and use that to generate a list of krynn locales.
+	 * Then we'll pick one of those at random and put it in the "location" field for the post.
+	 *
+	 */
+
+	$krynn = array('Ansalon','Abanasinia','Solace','Haven','Gateway','Qualinost','Ankatavaka','Pax Tharkas','Ergoth','Newsea','Straights of Schallsea','Plains of Dust','Tarsis','Barren Hills','Que Shu','Citadel of Light','Solinari','Hedge Maze','Tower of High Sorcery','Inn of the Last Home','Last Heroes Tomb','Academy of Sorcery','Gods Row','Temple of Majere','Temple of Kiri-Jolith','Temple of Mishakal','Temple of Zeboim,','The Trough','Sad Town','Xak Tsaroth','Zhaman','Skullcap','Saifhum','Karthay','Mithas','Kothas','Silver Dragon Mountain','Silvanesti');
+
+	$planet = array_rand($krynn,1);
+	$item['location'] = $krynn[$planet];
+
+	return;
+}
+
+
+
+
+/**
+ *
+ * Callback from the settings post function.
+ * $post contains the $_POST array.
+ * We will make sure we've got a valid user account
+ * and if so set our configuration setting for this person.
+ *
+ */
+
+function krynn_settings_post($a,$post) {
+	if(! local_user())
+		return;
+	if($_POST['krynn-submit'])
+		set_pconfig(local_user(),'krynn','enable',intval($_POST['krynn']));
+}
+
+
+/**
+ *
+ * Called from the Plugin Setting form. 
+ * Add our own settings info to the page.
+ *
+ */
+
+
+
+function krynn_settings(&$a,&$s) {
+
+	if(! local_user())
+		return;
+
+	/* Add our stylesheet to the page so we can make our settings look nice */
+
+	$a->page['htmlhead'] .= '<link rel="stylesheet"  type="text/css" href="' . $a->get_baseurl() . '/addon/krynn/krynn.css' . '" media="all" />' . "\r\n";
+
+	/* Get the current state of our config variable */
+
+	$enabled = get_pconfig(local_user(),'krynn','enable');
+
+	$checked = (($enabled) ? ' checked="checked" ' : '');
+
+	/* Add some HTML to the existing form */
+
+    $s .= '<span id="settings_krynn_inflated" class="settings-block fakelink" style="display: block;" onclick="openClose(\'settings_krynn_expanded\'); openClose(\'settings_krynn_inflated\');">';
+	$s .= '<h3>' . t('Krynn') . '</h3>';
+	$s .= '</span>';
+	$s .= '<div id="settings_krynn_expanded" class="settings-block" style="display: none;">';
+	$s .= '<span class="fakelink" onclick="openClose(\'settings_krynn_expanded\'); openClose(\'settings_krynn_inflated\');">';
+	$s .= '<h3>' . t('Krynn') . '</h3>';
+	$s .= '</span>';
+
+
+    $s .= '<div class="settings-block">';
+	$s .= '<h3>' . t('Krynn Settings') . '</h3>';
+	$s .= '<div id="krynn-enable-wrapper">';
+	$s .= '<label id="krynn-enable-label" for="krynn-checkbox">' . t('Enable Krynn Plugin') . '</label>';
+	$s .= '<input id="krynn-checkbox" type="checkbox" name="krynn" value="1" ' . $checked . '/>';
+	$s .= '</div><div class="clear"></div>';
```

Comment:
One thing left over (I think); in line 155 the div tag needs to be closed in line 166 after the end of the settings block. So line 166 should look like:

    $s .= '</div><div class="clear"></div></div>';

else the settings of all addons loaded after your addon will be hidden as the div block is not closed.